formio/formio.js
# FIO-6675/6676: Fixes issues where wizard page is getting removed regardless user confirmed removement or canceled it…

## Link to Jira Ticket
https://formio.atlassian.net/browse/FIO-6675
https://formio.atlassian.net/browse/FIO-6676

## Description

Previously, there were two different event listeners attached to the Remove componenent button in the form builder. One was attched orignially in the WebformBuilder and the second was attached to the panels in the WizardBuilder. The original methid is firing a confirmation dialog when the component has some childs and the one that in the Wizard builder was required to properly remove the page. The problem was that the event listener in the WizardBuilder had no idea about the confirmation window and user's answer. Moreover, when the user tried to remove a page from the component settings window, the second event listener was not fired at all, so the page was not removed properly.

Instead of attaching  two even listeners, I extended removeComponent method in the WizardBuilder.


## How has this PR been tested?

*I added automated tests to cover the following cases: 
1. Remove wizard page from the builder and confirm
2. Remove wizard page form the builder and cancel
3. Remove wizard page from the component settings and confirm
